assets/js/components/setup/CompatibilityChecks/checks.js

```diff
@@ -47,8 +48,8 @@ export const registryCheckSetupTag = ( registry ) => async () => {
 export const checkHealthChecks = async () => {
 	const response = await API.get( 'core', 'site', 'health-checks', undefined, {
 		useCache: false,
-	} ).catch( () => {
-		throw ERROR_FETCH_FAIL;
+	} ).catch( ( error ) => {
+		throw 'fetch_fail' === error.code ? ERROR_FETCH_FAIL : ERROR_API_UNAVAILABLE;
```

Comment:
The error code here needs to be `fetch_error` 

![image](https://user-images.githubusercontent.com/1621608/113834447-86a6d800-9793-11eb-8a2d-f27b70ef6527.png)